src/config/rcxml.c

```diff
@@ -1163,63 +1163,94 @@ rcxml_init(void)
 	rc.workspace_config.min_nr_workspaces = 1;
 }
 
-static struct {
-	const char *binding, *action, *attribute, *value;
+static struct key_combos {
+	const char *binding, *action;
+	struct {
+		const char *name, *value;
+	} attributes[3];
 } key_combos[] = {
-	{ "A-Tab", "NextWindow", NULL, NULL },
-	{ "W-Return", "Execute", "command", "alacritty" },
-	{ "A-F3", "Execute", "command", "bemenu-run" },
-	{ "A-F4", "Close", NULL, NULL },
-	{ "W-a", "ToggleMaximize", NULL, NULL },
-	{ "A-Left", "MoveToEdge", "direction", "left" },
-	{ "A-Right", "MoveToEdge", "direction", "right" },
-	{ "A-Up", "MoveToEdge", "direction", "up" },
-	{ "A-Down", "MoveToEdge", "direction", "down" },
-	{ "W-Left", "SnapToEdge", "direction", "left" },
-	{ "W-Right", "SnapToEdge", "direction", "right" },
-	{ "W-Up", "SnapToEdge", "direction", "up" },
-	{ "W-Down", "SnapToEdge", "direction", "down" },
-	{ "A-Space", "ShowMenu", "menu", "client-menu"},
-	{ "XF86_AudioLowerVolume", "Execute", "command", "amixer sset Master 5%-" },
-	{ "XF86_AudioRaiseVolume", "Execute", "command", "amixer sset Master 5%+" },
-	{ "XF86_AudioMute", "Execute", "command", "amixer sset Master toggle" },
-	{ "XF86_MonBrightnessUp", "Execute", "command", "brightnessctl set +10%" },
-	{ "XF86_MonBrightnessDown", "Execute", "command", "brightnessctl set 10%-" },
-	{ NULL, NULL, NULL, NULL },
+	{ "A-Tab", "NextWindow", {
+		{ NULL, NULL } }
+	}, { "W-Return", "Execute", {
+		{ "command", "alacritty" } }
+	}, { "A-F3", "Execute", {
+		{ "command", "bemenu-run" } }
+	}, { "A-F4", "Close", {
+		{ NULL, NULL } }
+	}, { "W-a", "ToggleMaximize", {
+		{ NULL, NULL } }
+	}, { "A-Left", "MoveToEdge", {
+		{ "direction", "left" } }
+	}, { "A-Right", "MoveToEdge", {
+		{ "direction", "right" } }
+	}, { "A-Up", "MoveToEdge", {
+		{ "direction", "up" } }
+	}, { "A-Down", "MoveToEdge", {
+		{ "direction", "down" } }
+	}, { "W-Left", "SnapToEdge", {
+		{ "direction", "left" } }
+	}, { "W-Right", "SnapToEdge", {
+		{ "direction", "right" } }
+	}, { "W-Up", "SnapToEdge", {
+		{ "direction", "up" } }
+	}, { "W-Down", "SnapToEdge", {
+		{ "direction", "down" } }
+	}, { "A-Space", "ShowMenu", {
+		{ "menu", "client-menu" },
+		{ "atCursor", "no" } }
+	}, { "XF86_AudioLowerVolume", "Execute", {
+		{ "command", "amixer sset Master 5%-" } }
+	}, { "XF86_AudioRaiseVolume", "Execute", {
+		{ "command", "amixer sset Master 5%+" } }
+	}, { "XF86_AudioMute", "Execute", {
+		{ "command", "amixer sset Master toggle" } }
+	}, { "XF86_MonBrightnessUp", "Execute", {
+		{ "command", "brightnessctl set +10%" } }
+	}, { "XF86_MonBrightnessDown", "Execute", {
+		{ "command", "brightnessctl set 10%-" } }
+	}, { NULL, NULL, { { NULL, NULL } } },
 };
 
 static void
 load_default_key_bindings(void)
 {
 	struct keybind *k;
 	struct action *action;
+	struct key_combos *current;
```

Comment:
Could reduce scope of `current` (put inside `(for (int i...) { ... }`

Reply:
Moved in this function and in `load_default_mouse_bindings`.